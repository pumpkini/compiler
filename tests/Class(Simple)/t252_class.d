class dynamic_array {
	int [] arr;
	int len;
	
	void init() {
		arr = NewArray(3, int);
		len = 0;
	}
	
	void append(int m) {
		if (len == arr.length()) {
			int i;
			int [] temp;
			temp = arr;
			arr = NewArray(2 * arr.length(), int);
			for(i = 0; i < len; i = i + 1) {
				arr[i] = temp[i];
			}
		}
		arr[len] = m;
		len = len + 1;
	}
	
	int pop() {
		int ret;
		if (len == 0) {
			return 0;
		}
		len = len - 1;
		ret = arr[len];
		if (3 * len < arr.length()) {
			int i;
			int [] temp;
			temp = arr;
			arr = NewArray(arr.length() / 2 + 1, int);
			for (i = 0;i < len;i = i + 1) {
				arr[i] = temp[i];
			}
		}
		return ret;
	}
	
	void print() {
		int i;
		for(i = 0;i < len;i = i + 1) {
			Print(i, "th element of array : ", arr[i]);
		}
	}
}


int main() {
	int j;
	dynamic_array t;
	t = new dynamic_array;
	t.init();
	for (j = 0;j < 40;j = j + 1) {
		if (j % 8 != 7) {
			t.append(j);
		} else {
			t.pop();
		}
		Print(j," elements of array : ");
		t.print();
	}
	
}
