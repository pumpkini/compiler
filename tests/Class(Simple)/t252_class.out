0 elements of array : 
0th element of array : 0
1 elements of array : 
0th element of array : 0
1th element of array : 1
2 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 6
7 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
8 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
9 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
10 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
11 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
12 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
13 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
14 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 14
15 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
16 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
17 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
18 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
19 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
20 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
21 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
22 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 22
23 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
24 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
25 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
26 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
27 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
28 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
29 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
30 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 30
31 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
32 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 32
33 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 32
25th element of array : 33
34 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 32
25th element of array : 33
26th element of array : 34
35 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 32
25th element of array : 33
26th element of array : 34
27th element of array : 35
36 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 32
25th element of array : 33
26th element of array : 34
27th element of array : 35
28th element of array : 36
37 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 32
25th element of array : 33
26th element of array : 34
27th element of array : 35
28th element of array : 36
29th element of array : 37
38 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 32
25th element of array : 33
26th element of array : 34
27th element of array : 35
28th element of array : 36
29th element of array : 37
30th element of array : 38
39 elements of array : 
0th element of array : 0
1th element of array : 1
2th element of array : 2
3th element of array : 3
4th element of array : 4
5th element of array : 5
6th element of array : 8
7th element of array : 9
8th element of array : 10
9th element of array : 11
10th element of array : 12
11th element of array : 13
12th element of array : 16
13th element of array : 17
14th element of array : 18
15th element of array : 19
16th element of array : 20
17th element of array : 21
18th element of array : 24
19th element of array : 25
20th element of array : 26
21th element of array : 27
22th element of array : 28
23th element of array : 29
24th element of array : 32
25th element of array : 33
26th element of array : 34
27th element of array : 35
28th element of array : 36
29th element of array : 37
